File: scripts/perl/post.pl
#!/usr/bin/perl

use strict;
use warnings;

use CGI;

# Create a new CGI object
my $cgi = CGI->new;

# Read image data from stdin
my $imageData;
while (<STDIN>) {
    $imageData .= $_;
}

# Set the "Content-Type" header for an image and the content length, ending with "\r\n"
print $cgi->header(-type => 'image/jpeg\r\n', -content_length => length($imageData)) . "\r\n\r\n";

# Output the image data
print $imageData;

File: scripts/python/post.py
#!/usr/bin/python

import sys
import os
import cgi

# Read image data from stdin
image_data = sys.stdin.read()

# Set the "Content-Type" header for an image and the content length, ending with "\r\n"
print("Content-Type: image/jpeg\r\n")
print("Content-Length: {}\r\n".format(len(image_data)))
print("\r\n")

# Output the image data
sys.stdout.write(image_data)
